### PR TITLE
MPAS applications: fixes for absolute vorticity and surface latent-heat flux

### LINKED ISSUE #1068: MPAS applications: issues with absolute vorticity and latent-heat flux fields
(1) The GRIB2 fields of "ABSV" (absolute vorticity) on isobaric surfaces are problematic when using UPP on HFIP–MPAS output.  For these ABSV fields: 

- A 1-DX "ring" of noisy values is present, and
- values are entirely zero within this ring.

The presence of missing values surrounding the computational domain is the underlying reason for this problem.  To fix the issue, the calculations for obtaining wind at mass points ("UH" and "VH") should first check for "SPVAL".

(2) Additionally, the surface latent-heat flux currently reads-in from a netCDF array called "QFX".  However, for current MPAS applications in RRFS and HFIP, this array is actually named "LH."

This issue can be assigned to @jaymes-kenyon

### PR DESCRIPTION
**Pull request by jaymes-kenyon**
This PR addresses issue #1068.  It has been tested in MPAS applications (GSL's RRFS and HFIP configurations).

**(1) Fix for absolute vorticity:**  Currently, in INITPOST_MPAS.F, the U and V winds are interpolated (averaged) onto mass points without checking for missing values.  However, for the HFIP domain, a region of missing values surrounds the computational domain.  As a result, the UPP interpolation procedure at the domain edge is contaminated by adjacent "SPVAL" grid points.  Moreover, the subsequent calculations involving horizontal U and V gradients (e.g., for absolute vorticity, or "ABSV") yield very large and "noisy" values at the domain edge.  The GRIB2 compression struggles with these "ABSV" fields; see attached screenshot of 850-hPa absolute vorticity (left panel).

This PR ensures that U and V winds are interpolated onto mass points only after checking for the presence of "SPVAL" at adjacent grids.  Additionally, the "ABSV" arrays are initialized with "SPVAL", regardless of MODELNAME.  The corrected field of 850-hPa absolute vorticity is shown in the attached screenshot (right panel).

absv

**(2) Fix for surface latent-heat flux:** This field is now read-in from the netCDF "LH" array

### CODE REVIEW

**Comment by jaymes-kenyon** [comment]
Converting this PR back to a draft — I will likely add another commit to fix an unrelated issue.

File: sorc/ncep_post.fd/UPP_PHYSICS.f [hunk 1]
```diff
@@ -1773,21 +1774,12 @@ SUBROUTINE CALVOR(UWND,VWND,ABSV)
 !     
 !     LOOP TO COMPUTE ABSOLUTE VORTICITY FROM WINDS.
 !     
-      IF(MODELNAME  == 'RAPR') then
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/UPP_PHYSICS.f:1776]: @jaymes-kenyon The changes will have impact on results of rap and hrrr. There might be an upcoming hrrr implementation to include GTG4 updates. I suggest keeping the current settings for rap and hrrr. You may initialize ABSV as SPVAL when modelname is 'RAPR' and submodelname is 'MPAS'.

**Comment by WenMeng-NOAA** [comment]
@gspetro-NOAA @FernandoAndrade-NOAA You may start the UPP RTs on R&D machines. There should be no baseline changes with this PR.

**Comment by WenMeng-NOAA** [comment]
The UPP RTs have been completed on WCOSS2 with no baseline changes.

**Comment by WenMeng-NOAA** [comment]
@jaymes-kenyon Please take a look at my UPP standalone test with MPAS_hfip data at
/home/Wen.Meng/stmp2/mpas_2024080900 on Hera
I see ABSV corrected as
```
441:789049528:ndata=5435001:undef=2423703:mean=6.4884e-05:min=-0.00244:max=0.00308:cos_wt_mean=6.35469e-05:d=2024080812:ABSV:975 mb:12 hour fcst:
452:810255441:ndata=5435001:undef=2423703:mean=6.49881e-05:min=-0.00241:max=0.00327:cos_wt_mean=6.365e-05:d=2024080812:ABSV:1000 mb:12 hour fcst:
```

**Comment by jaymes-kenyon** [comment]
@WenMeng-NOAA — Your test looks good—thanks for sharing!

**Review by gspetro-NOAA — APPROVED** [review]
RTs passed on Hera, Orion, and Hercules with no baseline changes.

**Review by WenMeng-NOAA — APPROVED** [review]

**Comment by WenMeng-NOAA** [comment]
This PR is ready for merging.

**Pull request merged**